runtime/caml/domain_state.tbl

```diff
@@ -14,8 +14,8 @@
 /*                                                                        */
 /**************************************************************************/
 
-DOMAIN_STATE(value*, young_ptr)
 DOMAIN_STATE(value*, young_limit)
+DOMAIN_STATE(value*, young_ptr)
```

Comment:
Just curious: why the permutation of `young_ptr` and `young_limit` in the domain state?  Does it make a difference?

Reply:
This means we don't need an offset when loading using the register containing the `caml_state` pointer.

On amd64 it saves us a byte per poll.

Reply:
I ran the numbers on this overnight for amd64 (there should be no difference on arm64). I compared https://github.com/sadiqj/ocaml/tree/polling_flip_limit_ptr and this PR. In this case `polling_flip` confusingly refers to flipping things _back_ to how they are on trunk.

It seems shuffling the two fields has a surprisingly large impact on code size. Looking at the results it occurred to me that this change impacts allocations as well, they get a byte shorter.

&nbsp; | 4.13.0+polling | 4.13.0+polling_flip
-- | -- | --
code size incr (%) | 1.010848 | 1.016079

[Raw benchmark results](https://github.com/sadiqj/sandmark_benchmark_data/tree/main/young_ptr_flip/amd64)

Reply:
OK, thanks for the explanation.  That's a good trick for x86-64!